crypto/store/loader_file.c

```diff
@@ -0,0 +1,1069 @@
+            }
+        }
+
+        if (*matchcount > 1)
+            OSSLSTOREerr(OSSLSTORE_F_FILE_LOAD_TRY_DECODE,
+                         OSSLSTORE_R_AMBIGUOUS_CONTENT_TYPE);
+        if (*matchcount == 0)
+            OSSLSTOREerr(OSSLSTORE_F_FILE_LOAD_TRY_DECODE,
+                         OSSLSTORE_R_UNSUPPORTED_CONTENT_TYPE);
+        else if (matching_handlers[0]->repeatable) {
+            if (ctx == NULL) {
+                OSSLSTOREerr(OSSLSTORE_F_FILE_LOAD_TRY_DECODE,
+                             OSSLSTORE_R_UNSUPPORTED_CONTENT_TYPE);
+                OSSLSTORE_INFO_free(result);
+                result = NULL;
+            } else {
+                ctx->_.file.last_handler = matching_handlers[0];
+                ctx->_.file.last_handler_ctx = handler_ctx;
+            }
+        }
+
+        OPENSSL_free(matching_handlers);
+    }
+
+ err:
+    OPENSSL_free(new_pem_name);
+    BUF_MEM_free(new_mem);
+
+    if (result != NULL
+        && (t = OSSLSTORE_INFO_get_type(result)) == OSSLSTORE_INFO_EMBEDDED) {
+        pem_name = new_pem_name = osslstore_info_get0_EMBEDDED_pem_name(result);
+        new_mem = osslstore_info_get0_EMBEDDED_buffer(result);
+        data = (unsigned char *)new_mem->data;
+        len = new_mem->length;
+        OPENSSL_free(result);
+        result = NULL;
+        goto again;
+    }
+
+    if (result != NULL)
+        ERR_clear_error();
+
+    return result;
+}
+
+static OSSLSTORE_INFO *file_load_try_repeat(OSSLSTORE_LOADER_CTX *ctx,
+                                            const UI_METHOD *ui_method,
+                                            void *ui_data)
+{
+    OSSLSTORE_INFO *result = NULL;
+
+    if (ctx->_.file.last_handler != NULL) {
+        result =
+            ctx->_.file.last_handler->try_decode(NULL, NULL, NULL, 0,
+                                                 &ctx->_.file.last_handler_ctx,
+                                                 ui_method, ui_data);
+
+        if (result == NULL) {
+            ctx->_.file.last_handler->destroy_ctx(&ctx->_.file.last_handler_ctx);
+            ctx->_.file.last_handler_ctx = NULL;
+            ctx->_.file.last_handler = NULL;
+        }
+    }
+    return result;
+}
+
+static int file_read_pem(BIO *bp, char **pem_name, char **pem_header,
+                         unsigned char **data, long *len,
+                         const UI_METHOD *ui_method,
+                         void *ui_data)
+{
+    int i = PEM_read_bio(bp, pem_name, pem_header, data, len);
+
+    if (i <= 0)
+        return 0;
+
+    /*
+     * 10 is an arbitrary number to make a minimum check if the
+     * header is worth caring about.  The number is picked from
+     * crypto/pem/pem_info.c and has no explanation...
+     * A possible guess, though, is that the header keys that
+     * PEM_get_EVP_CIPHER_INFO() recognises are about 10 chars
+     * long, so anything shorter than that won't be recognised
+     * and is therefore not worth spending cycles on.
+     */
```

Comment:
It looks to me like 10 is the exact length of "Proc-Type:" that PEM_get_EVP_CIPHER_INFO() is expecting to see in the header. That function will fail if it doesn't see that (unless the header length is 0 or just contains "\n".